google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/sdk/CloudSdkPanel.java

```diff
@@ -70,46 +71,91 @@ public CloudSdkPanel() {
         .addDocumentListener(new DocumentAdapter() {
           @Override
           protected void textChanged(DocumentEvent event) {
-            checkSdk();
+            checkSdkInBackground();
           }
         });
 
-    checkSdk();
+    checkSdkInBackground();
   }
 
-  private void checkSdk() {
-    String path = cloudSdkDirectoryField.getText();
+  private void checkSdkInBackground() {
```

Comment:
I've manually tested, and this isn't an issue in that case. Since all UI updates we perform happen within calls to `Application.invokeLater`, intellij handles making sure we aren't updating anything that's in an inoperable state. Also, note that the method explicitly handles the case when the cloudsdk path is null.

I will do some small refactoring to hopefully make it more obvious that we won't run into problems if the panel is closed before the thread completes.